core/primitives/src/hexdisplay.rs

```diff
@@ -21,28 +21,37 @@ pub struct HexDisplay<'a>(&'a [u8]);
 
 impl<'a> HexDisplay<'a> {
 	/// Create new instance that will display `d` as a hex string when displayed.
-	pub fn from(d: &'a dyn AsBytesRef) -> Self { HexDisplay(d.as_bytes_ref()) }
+	pub fn from<R: AsBytesRef>(d: &'a R) -> Self { HexDisplay(d.as_bytes_ref()) }
```

Comment:
Why did you change the signature?

Reply:
Don't see any good reason for dynamic dispatch here